src/hevm/src/EVM/Fetch.hs

```diff
@@ -167,23 +167,22 @@ oracle state info q = do
          case noJump of
             -- Unsat means condition
             -- cannot be nonzero
-            -- We assume it to be zero in this case.
-            -- Theoretically, it is possible that our pathconditions
-            -- are inconsistent, either due to:
-            -- a) a false assumption
-            -- b) a previous timeout
-            -- We allow ourselves to ignore these cases with the motivation that
-            -- a) should be checked elsewhere and,
-            -- if b) occurs, we are in a unreachable path, and its better
-            -- to not keep on branching anyway.
-            Unsat -> return $ continue (EVM.Known 0)
+            Unsat -> do jump <- checksat $ pathconds .&& jumpcondition .== 0
+                        -- can it be zero?
+                        case jump of
+                          -- No. We are on an inconsistent path.
+                          Unsat -> return $ continue EVM.Inconsistent
+                          -- Yes. It must be 0.
+                          Sat -> return $ continue (EVM.Iszero True)
+                          -- Assume 0 is still possible.
+                          Unk -> return $ continue (EVM.Iszero True)
             -- Sat means its possible for condition
             -- to be nonzero.
             Sat -> do jump <- checksat $ pathconds .&& jumpcondition .== 0
                       -- can it also be zero?
                       case jump of
                         -- No. It must be nonzero
-                        Unsat -> return $ continue (EVM.Known 1)
+                        Unsat -> return $ continue (EVM.Iszero False)
```

Comment:
If I understood you correctly, this was another soundness bug @MrChico?

Could we add a test case that covers this behaviour?

Reply:
I discovered it through [this test](https://github.com/ethereum/solidity/blob/develop/test/libsolidity/smtCheckerTests/operators/compound_assignment_division_1.sol)

Reply:
Which was previously not a problem because we weren't storing the path condition information upon branching when the query was Known, but now we are adding this information anyway so that test should be sufficient 👍